### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by mciparelli**

### CODE REVIEW

File: bsconfig.json [hunk 1]
```diff
@@ -1,14 +1,19 @@
...
+    {
+      "dir": "tests",
+      "type": "dev"
     }
   ],
   "refmt": 3,
   "suffix": ".bs.js",
   "package-specs": {
-    "module": "es6",
+    "module": "commonjs",
```
Comment (jkrehm) [inline bsconfig.json]: Why switch from ES6 to CommonJS?

Comment (mciparelli) [reply bsconfig.json]: Built versions only matter for tests, and I just wanted to have it running as quickly as possible. I could've used no bundler, but the package I'm using to run tests in the browser adds the stdin as a normal script tag (https://github.com/juliangruber/browser-run/blob/master/static/index.html) so I needed to use a bundler and went with browserify which is the simplest to setup. If I wanted to keep es6 I would've had to use a browserify transform, felt simpler/faster to just switch to commonjs instead.

**Comment by mciparelli** [comment]
@jkrehm see if it got fixed now.

File: package.json [hunk 2]
```diff
@@ -19,18 +19,20 @@
     "build": "bsb -make-world",
     "clean": "bsb -clean-world",
     "start": "bsb -make-world -w",
-    "pretest": "rimraf tests/main.bs.js && yarn build && sed -i '1s;^;require(\"@babel/polyfill\");\n' tests/main.bs.js",
-    "test": "browserify tests/main.bs.js | tape-run --browser chrome --render=faucet"
+    "pretest": "rimraf tests/main.bs.js && yarn build && prepend tests/main.bs.js 'require(\"abortcontroller-polyfill/dist/abortcontroller-polyfill-only\");require(\"@babel/polyfill\");\n'",
```
Comment (jkrehm) [inline package.json]: Much closer. To get it working I had to change `pretest` to:
```javascript
rimraf tests/main.bs.js && yarn build && prepend tests/main.bs.js \"require('abortcontroller-polyfill/dist/abortcontroller-polyfill-only');require('@babel/polyfill');\n\"
```
The quotes were breaking things.

File: package.json [hunk 3]
```diff
@@ -19,18 +19,20 @@
...
+    "test": "browserify tests/main.bs.js | tape-run --render=faucet"
```
Comment (jkrehm) [inline package.json]: `faucet` is resulting in the following error:
```
$ browserify tests/main.bs.js | tape-run --render=faucet
events.js:183
      throw er; // Unhandled 'error' event
      ^

Error: spawn faucet ENOENT
    at _errnoException (util.js:992:11)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:190:19)
    at onErrorNT (internal/child_process.js:372:16)
    at _combinedTickCallback (internal/process/next_tick.js:138:11)
    at process._tickCallback (internal/process/next_tick.js:180:9)
    at Function.Module.runMain (module.js:695:11)
    at startup (bootstrap_node.js:191:16)
    at bootstrap_node.js:612:3
error Command failed with exit code 1.
```
Maybe a different renderer? Not setting one at all displays fine for me, but maybe my terminal client works differently.

**Pull request merged**